libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -288,6 +297,27 @@ void AP_ICEngine::update(void)
         }
     }
 
+    // check against redline RPM
+    if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        // update the filtered RPM value
+        filtered_rpm_value =  _rpm_filter.apply(rpm_value, AP::scheduler().get_loop_period_s());
```

Comment:
need to apply filter on every loop, and setup loop time once on filter
need to check rpm_instance is valid

Reply:
Fixed both. Also changed the cutoff frequency to 0.5 Hz as discussed.